profiles/seccomp/seccomp.go

```diff
@@ -37,38 +39,99 @@ func setupSeccomp(config *types.Seccomp) (newConfig *specs.Seccomp, err error) {
 		return nil, nil
 	}
 
-	newConfig = &specs.Seccomp{}
+	newConfig := &specs.Seccomp{}
+
+	var arch string
+	var native, err = libseccomp.GetNativeArch()
+	if err == nil {
+		arch = native.String()
+	}
+
+	if len(config.Architectures) != 0 && len(config.ArchMap) != 0 {
+		return nil, errors.New("'architectures' and 'archMap' were specified in the seccomp profile, use either 'architectures' or 'archMap'")
+	}
 
 	// if config.Architectures == 0 then libseccomp will figure out the architecture to use
-	if len(config.Architectures) > 0 {
-		for _, arch := range config.Architectures {
-			newConfig.Architectures = append(newConfig.Architectures, specs.Arch(arch))
+	if len(config.Architectures) != 0 {
+		for _, a := range config.Architectures {
+			newConfig.Architectures = append(newConfig.Architectures, specs.Arch(a))
+		}
+	}
+
+	if len(config.ArchMap) != 0 {
+		for _, a := range config.ArchMap {
+			if string(a.Arch) == arch {
```

Comment:
I will look at adding a test in a separate PR as it is unrelated to this. I think it is possible it just needs some architecture conditionals...